apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/TableStatsCollectorUtil.java

```diff
@@ -627,6 +672,659 @@ public static List<CommitEventTablePartitions> populateCommitEventTablePartition
+    if (rows.isEmpty()) {
+      log.warn("No data found in data_files for table: {}", fullTableName);
+      return null;
+    }
+
+    return rows.get(0);
+  }
+
+  /**
+   * Build CommitMetadata object from an Iceberg Snapshot.
+   *
+   * @param snapshot Iceberg snapshot
+   * @return CommitMetadata with snapshot information
+   */
+  private static CommitMetadata buildCommitMetadataFromSnapshot(Snapshot snapshot) {
+    Long snapshotId = snapshot.snapshotId();
+    Long committedAt = snapshot.timestampMillis();
+    String operation = snapshot.operation();
+    Map<String, String> summary = snapshot.summary();
+
+    CommitOperation commitOp = null;
+    if (operation != null) {
+      try {
+        commitOp = CommitOperation.valueOf(operation.toUpperCase());
+      } catch (IllegalArgumentException e) {
+        log.warn("Unknown commit operation: {}", operation);
+      }
+    }
+
+    return CommitMetadata.builder()
+        .commitId(snapshotId)
+        .commitTimestampMs(committedAt)
+        .commitAppId(summary.getOrDefault("spark.app.id", "unknown"))
+        .commitAppName(summary.getOrDefault("spark.app.name", "unknown"))
+        .commitOperation(commitOp)
+        .build();
+  }
+
+  /**
+   * Build CommitEventTablePartitionStats object from extracted data.
+   *
+   * @param table Iceberg table
+   * @param spark SparkSession
+   * @param schema Table schema
+   * @param statsRow Row with aggregated statistics
+   * @param commitMetadata Commit metadata
+   * @param metricsMap Map of column-level metrics
+   * @param partitionData List of partition data (empty for unpartitioned tables)
+   * @return CommitEventTablePartitionStats object, or null if table metadata extraction fails
+   */
+  private static CommitEventTablePartitionStats buildPartitionStatsObject(
+      Table table,
+      SparkSession spark,
+      Schema schema,
+      Row statsRow,
+      CommitMetadata commitMetadata,
+      Map<String, List<ColumnData>> metricsMap,
+      List<ColumnData> partitionData) {
+
+    String fullTableName = table.name();
+    String dbName = getDatabaseName(fullTableName);
+    if (dbName == null) {
+      return null;
+    }
+
+    TableIdentifier identifier = TableIdentifier.parse(fullTableName);
+    String tableName = identifier.name();
+    String clusterName = getClusterName(spark);
+    String tableMetadataLocation = table.location();
+    String partitionSpecString = table.spec().toString();
+
+    // Extract stats from row
+    Long rowCount = statsRow.getAs("total_row_count");
+    Long columnCount = (long) schema.columns().size();
+
+    // Extract column-level metrics from map
+    List<ColumnData> nullCounts = metricsMap.get("nullCount");
+    List<ColumnData> nanCounts = metricsMap.get("nanCount");
+    List<ColumnData> minValues = metricsMap.get("minValue");
+    List<ColumnData> maxValues = metricsMap.get("maxValue");
+    List<ColumnData> columnSizes = metricsMap.get("columnSize");
```

Comment:
@xkrogen You might be interested in this PR. Let me also share doc from DQ team offline.